feat: 그룹 상세 화면 UI 구현

HJunng opened the pull request:
## 📝 변경 사항
그룹 상세 화면의 UI를 구현하고, 홈 화면에서 그룹 클릭 시 상세 화면으로 이동하는 네비게이션 기능을 추가했습니다.

## 🎯 작업 내용
- 그룹 상세 화면 UI 구현
  - 갤러리 탭과 지도 탭을 가진 탭 레이아웃 구현
  - TopAppBar에 뒤로가기, 검색, 메뉴 버튼 추가
  - Staggered Grid를 활용한 갤러리 뷰 구현
- 네비게이션 구조 개선
  - `GroupDetail` Route 추가
  - 홈 화면에서 그룹 클릭 시 상세 화면으로 이동하는 네비게이션 로직 구현
  - `MemoripNavigator`에 `navigateToRoute` 메서드 추가
- Staggered Grid 컴포넌트 개선
  - `PlaceImageItem` 모델로 이미지 데이터 구조화
  - 이미지 비율을 고려한 `aspectRatio` 적용
  - 이미지 placeholder 색상 개선
- 리소스 추가
  - 아이콘 추가 (뒤로가기, 메뉴, 검색)
  - 그룹 상세 화면 관련 문자열 리소스 추가

## 🔗 관련 이슈
Closes #15

## 📸 스크린샷 (선택사항)
<img width="50%" alt="image" src="https://github.com/user-attachments/assets/3406a87f-9f80-4b82-a7e8-b910a491473e" />


## 💬 리뷰어에게
- 갤러리 탭은 Staggered Grid로 구현했고, 지도 탭은 내용이 길어져서 다른 브랜치에서 작업중입니다.
- 현재는 더미 이미지 데이터를 사용하고 있습니다.

HJunng committed:
feat: 그룹 상세 화면 구현

HJunng committed:
feat: 그룹 상세 화면 구현

HJunng committed:
feat: PlaceImageItem 모델 생성

HJunng committed:
feat: PlaceImageItem 모델 생성

HJunng committed:
feat: PlaceImageItem에 맞춰서 StaggeredGrid 레이아웃 수정

HJunng committed:
feat: PlaceImageItem에 맞춰서 StaggeredGrid 레이아웃 수정

HJunng committed:
feat: TabView로 보여줄 GalleryTab, MapTab 추가

HJunng committed:
feat: TabView로 보여줄 GalleryTab, MapTab 추가

HJunng committed:
feat: 아이콘 추가

HJunng committed:
feat: 아이콘 추가

HJunng committed:
feat: 그룹상세화면 TopAppBar 구현

HJunng committed:
feat: 그룹상세화면 TopAppBar 구현

HJunng committed:
fix: 이미지 placeholder 색상 변경

HJunng committed:
fix: 이미지 placeholder 색상 변경

HJunng committed:
Merge remote-tracking branch 'origin/feature/15-group-detail' into feature/15-group-detail

# Conflicts:
#	client/presentation/src/main/res/values/strings.xml

HJunng committed:
feat: 그룹 상세 화면으로 이동하는 nav route 추가

HJunng committed:
feat: 그룹 상세 화면 내비게이션 추가

- 홈 화면에서 그룹 클릭 시 그룹 상세 화면으로 이동하는 내비게이션 로직 추가
- `groupDetail` 컴포저블을 `MemoripNav`에 추가

HJunng committed:
feat: 그룹 클릭 이벤트 핸들러 추가

- `HomeScreen`과 `HomeNavGraph`에 그룹 클릭 시 호출될 `onGroupClick` 콜백 추가

HJunng committed:
feat: 그룹 상세 화면 NavGraph 및 Content 분리

- `GroupDetailScreen`과 `GroupDetailScreenContent`를 분리
- 그룹 상세 화면의 NavGraph Entry point `groupDetail` 추가
- 임시 이미지 데이터를 생성하여 `GroupDetailScreen`에 적용
- @Preview의 이름을 GroupDetailScreenContent로 변경

uselessnaming commented on client/presentation/src/main/java/com/andone/memorip/presentation/component/MemoripStaggeredGrid.kt:
```diff
@@ -35,17 +34,24 @@ private object StaggeredGridDimens {
         verticalItemSpacing = MemoripSpace.SpaceXXSmall,
         horizontalArrangement = Arrangement.spacedBy(MemoripSpace.SpaceXXSmall),
         modifier = modifier.fillMaxSize(),
         content = {
```
content slot은 원래 하던대로 {}로 사용해도 좋을 것 같아요!

uselessnaming reviewed (approved):
확인했습니다! 추가적인 부분은 없는 것 같아서 전체 흐름 확인하고 갑니다!

GwonDongHyeon21 commented on client/presentation/src/main/java/com/andone/memorip/presentation/groupdetail/GroupDetailScreen.kt:
```diff
@@ -0,0 +1,181 @@
+    )
+    val coroutineScope = rememberCoroutineScope()
+
+    Scaffold(
+        topBar = {
+            GroupDetailAppBar(
+                title = groupName,
+                onBackClick = onBackClick,
+                onMenuClick = onMenuClick,
+                onSearchClick = onSearchClick
+            )
+        },
+        modifier = modifier
+    ) { innerPadding ->
```
선언 순서가 modifier가 topBar보다 위인걸로 압니다!

GwonDongHyeon21 reviewed (approved):
고생하셨습니다. 더미데이터 util 패키지에서 한번에 관리해서 재사용하면 좋을 것 같습니다!
AppBar를 component에 둘지, 한 파일에서 section처럼 분리만 할지 이야기 해보면 좋을 것 같습니다. 네이밍도 맞추면 좋을 것 같네요!

GwonDongHyeon21 commented on client/presentation/src/main/java/com/andone/memorip/navigation/MemoripNavigator.kt:
```diff
@@ -32,6 +32,11 @@ class MemoripNavigator(
+        if (currentTab?.route == route) return
+        backStack.add(route)
+    }
+
```
저는 각 화면별로 이동하는 함수를 개별적으로 만들어서, 각 화면 NavGraph에 확장함수로 작성했습니다. 분리해서 작성하면 각 화면 이동별로 설정을 따로 할 수 있어서 더 좋다고 생각했습니다!

HJunng committed:
Merge remote-tracking branch 'origin/dev' into feature/15-group-detail

# Conflicts:
#	client/presentation/src/main/java/com/andone/memorip/navigation/MemoripNavigator.kt
#	client/presentation/src/main/java/com/andone/memorip/navigation/Route.kt

HJunng committed:
feat: 그룹 상세 화면 네비게이션 로직 분리

- `MemoripNavigator`에서 `navigateToRoute` 함수를 제거하고, `navigateToGroupDetail` 함수로 변경
- `navigateToGroupDetail`을 확장 함수로 분리, `NavBackStack`에서 직접 호출할 수 있도록 변경
- 홈 화면에서 그룹 클릭 시 `navigateToGroupDetail`을 호출하도록 수정

HJunng committed:
feat: 프리뷰용 더미 데이터 관리 객체 추가

- 프리뷰에서 `DummyData`를 사용하도록 수정했습니다.
